src/index.js

```diff
@@ -68,21 +68,27 @@ class GoogleCloudStorageAdapter extends BaseAdapter {
   }
 
   read(options = {}) {
-    const readStream = this.bucket.file(options.path).createReadStream()
+    const prefix = `https://${this.assetDomain}/`
+    let path = options.path
+    if (path.indexOf(prefix) === 0) {
+      path = path.slice(prefix.length);
+    }
+    
+    const readStream = this.bucket.file(path).createReadStream()
     let contents = null;
     return new Promise(function(resolve, reject) {
       readStream.on('error', function(err) {
         return reject(err);
       });
       readStream.on('data', function(data) {
         if (contents) {
-          contents = data;
-        } else {
           contents = Buffer.concat([contents, data]);
+        } else {
+          contents = data;
         }
       });
       readStream.on('end', function() {
-        return resolve(content);
+        return resolve(contents);
```

Comment:
Nice catch! How could this even work prior to this PR... 🤔

Reply:
The `read()` function seems rarely called. The function is used by Ghost to serve favicon, not when serving images.